feature/gribi/otg_tests/gribi_programming_performance/README.md

```diff
@@ -0,0 +1,54 @@
+# TE-14.1: gRIBI Scaling
+
+## Summary
+
+Validate and measure gRIBI programming performance for a DUT.
+
+## Topology
+
+```
+                      _________
+                     |         |
+   [ ATE p1 ] --- p1 |   DUT   | p2 --- [ ATE p2 ]     
+                     |_________|
+```
+
+ * ATE p1 to DUT p1 is assigned `192.0.2.0/30` and `2001:db8::/127`.
+ * ATE p2 to DUT p2 is assigned `192.0.2.2/30` and `2001:db8::2/127`.
+
+ * Address pools that are injected to the DUT by the test should correspond to Internet DFZ route distributions.
+    * Pool sizes to be tested:
+      * 1,000
+      * 10,000
+      * 100,000
+      * 500,000
+      * 1,000,000
+
+## Procedure
+
+ * Utilising the topology described above, configure DUT and ATE.
+ * Establish a gRIBI connection to the DUT, using `FIB_ACK`, `SINGLE_PRIMARY` and `PERSIST` session parameters.
+ * For each of the pool sizes defined above, referred to as N:
+  - Select a set of N routes from the DFZ.
+  - Order the set, and select the 25th, 50th, 75th, 90th, 95th, 100th percentile entries.
+  - Define traffic flows destined to the prefixes selected above.
+  - Begin generating traffic.
+  - Program all entries into the DUT:
+    - Record the programming latency for each prefix for both RIB and FIB ACKs to be received.
```

Comment:
Should we program before generating traffic?

Reply:
Per the logic here, I'm saying that we have two means of measuring programming latency:

 * the reported operation latency within the test (this can use a time that we send the operation vs. the [`timestamp`](https://github.com/openconfig/gribi/blob/master/v1/proto/service/gribi.proto#L308) that is reported in the `AFTResult` that is returned.
 *  the amount of loss that we observe for the prefixes that we are monitoring above.

We have to start traffic before programming to ensure that we're already sending packets before we start measuring.

Does this make sense? I can include more verbose text if needed.

Reply:
Ah this makes sense, I actually missed that this was for programming latency. I for some reason thought it was about scale.

Reply:
Ah, I think it's both -- it validates that we can hit X scale, as well as then determine the latency of the programming.